testPad/compound.testpad.test.ts

```diff
@@ -0,0 +1,329 @@
+        4_000_000_000
+      ),
+      alice
+    );
+  }
+
+  it("auto-compound: schedule & execute task", async () => {
+    testLog.getLog().info("running section: prepare compound extrinsic");
+
+    // @ts-ignore
+    const proxy = await oakApi.api.rpc.xcmpHandler.crossChainAccount(
+      user1.keyRingPair.address
+    );
+    const assetId = await getNextAssetId();
+    const poolId = assetId.add(BN_ONE);
+    await Sudo.batchAsSudoFinalized(
+      Assets.mintNative(user1),
+      Assets.issueToken(user1),
+      Sudo.sudoAs(user1, api.tx.proxy.addProxy(proxy, { AutoCompound: 0 }, 0)),
+      Sudo.sudoAs(
+        user1,
+        Xyk.createPool(BN_ZERO, BN_MILLION, assetId, BN_MILLION)
+      )
+    );
+
+    // it is complex to set up the chain with predefined rewards, it is enough to test without it,
+    // we add 1_000 MGR into the poolId
+    const tx = api.tx.proxy.proxy(
+      user1.keyRingPair.address,
+      undefined,
+      api.tx.xyk.provideLiquidityWithConversion(poolId, 0, BN_THOUSAND)
+    );
+    const encodedTxHex = tx.unwrap().toHex();
+    const encodedTxInfo = await tx.paymentInfo(user1.keyRingPair);
+
+    testLog.getLog().info(encodedTxHex);
+    testLog.getLog().info("running section: schedule task: " + encodedTxHex);
+
+    // @ts-ignore
+    const taskIdLog = await oakApi.api.rpc.automationTime.generateTaskId(
+      alice.keyRingPair.address,
+      "compound"
+    );
+    testLog.getLog().info(`task id: ${taskIdLog}`);
+    const taskId = "compound";
+    // const seconds = Math.trunc(new Date().getTime() / 1000);
+    // const nextHour = seconds - (seconds % HOUR) + HOUR;
+
+    const executions = [0];
+    const txCreate = oakApi.api.tx.automationTime.scheduleXcmpTask(
+      // task id
+      oakApi.api.createType("Vec<u8>", taskId),
+      // intervals for task execution, 0 == immediate execution, only with 'features=dev_queue' oak build
+      { Fixed: { executionTimes: executions } },
+      // destination para id
+      2110,
+      // currency id - the registered TUR in this case
+      1,
+      // encoded call
+      oakApi.api.createType("Vec<u8>", encodedTxHex),
+      // extrinsic weight
+      encodedTxInfo.weight
+    );
+
+    const info = await txCreate.paymentInfo(user1.keyRingPair);
+    const currencyData =
+      await oakApi.api.query.xcmpHandler.xcmChainCurrencyData(2110, 1);
+    const totalWeight = new BN(encodedTxInfo.weight).add(
+      // @ts-ignore
+      new BN(currencyData.unwrap().instructionWeight)
+    );
+    const taskExecutionFee = totalWeight
+      // @ts-ignore
+      .mul(new BN(currencyData.unwrap().feePerSecond))
+      .div(weightInSeconds);
+
+    // const xcmExecutionFee = new BN(4_000_000_000) // TUR base unit cost weight * 4 instructions in transfer
+    //   .mul(new BN(419000000000)) // TUR fps
+    //   .div(weightInSeconds);
+
+    const automationTimeFee =
+      // @ts-ignore
+      await oakApi.api.rpc.automationTime.getTimeAutomationFees("XCMP", 1);
```

Comment:
The parameter of this call is incomplete yet. We will create a new call and demo code for you to return the correct fee estimate, but we need to decide the function to be triggerd by XCM, aka the `ClaimSwapDeposit`. Please refer to the above comment.